k2/csrc/tensor_ops_test.cu

```diff
@@ -325,10 +326,69 @@ static void TestSimpleRaggedIndexSelect1D() {
   }
 }
 
+
 TEST(Index, SimpleRaggedIndexSelect1D) {
   TestSimpleRaggedIndexSelect1D<float>();
   TestSimpleRaggedIndexSelect1D<double>();
   TestSimpleRaggedIndexSelect1D<int32_t>();
 }
 
+
+TEST(Tensor, DiscountedCumSum) {
+  for (int32_t i = 0; i < 4; i++) {
+    int32_t M = RandInt(0, 1000),
+        T = RandInt(0, 2000);  // TODO: increase.
+    while (M * T > 10000) {  // don't want test to take too long.
+      M /= 2;
+      T /= 2;
+    }
+
+    ContextPtr cuda_context = GetCudaContext(),
+        cpu_context = GetCpuContext();
+
+    Array2<float> x = RandUniformArray2<float>(cuda_context, M, T, -2.0, 2.0);
+    Array2<float> gamma = RandUniformArray2<float>(cuda_context, M, T, 0.0, 1.0);
+    Array2<float> y(cuda_context, M, T);
+    y = -10.0;
+
+    bool flip = (i % 2 == 1);
+
+    Array2<float> x_cpu = x.To(cpu_context),
+        gamma_cpu = gamma.To(cpu_context),
+        y_cpu(cpu_context, M, T);
+
+    Tensor x_ten = x.ToTensor(),
```

Comment:
I would suggest using

https://github.com/k2-fsa/k2/blob/0bcc330db4b566a22a431e692b10880a82c4b1ed/k2/csrc/tensor_ops_test.cu#L59-L60

to generate tensors with stride0 != 1.

The current test covers only a single case: `stride0 == 1`.

Reply:
Because it's flipping, it's testing stride1 == -1 as well...